asr/wenet/cli/reverb.py

```diff
@@ -0,0 +1,392 @@
+            )
+            self.model.encoder.global_cmvn = global_cmvn
+        self.model = self.model.to(self.device)
+        self.model.eval()
+
+        self.input_frame_length = self.test_conf["fbank_conf"]["frame_shift"]
+        frame_downsampling_factor = {"linear": 1, "conv2d": 4, "conv2d6": 6, "conv2d8": 8}
+        self.output_frame_length = self.input_frame_length * frame_downsampling_factor.get(
+            self.configs["encoder_conf"]["input_layer"], 4
+        )
+
+    def _make_path_absolute(
+        self,
+        config_path: str,
+        alternate_path: str | None = None
+    ):
+        """Returns the 'alternate_path' if set, otherwise get's the absolute path todo
+        'config_path'
+        """
+        if alternate_path:
+            return alternate_path
+
+        config_path = Path(config_path)
+        if not config_path.is_absolute():
+            # Assume it's adjacent to the model
+            checkpoint_parent = Path(self.checkpoint).parent
+            config_path = checkpoint_parent / config_path
+
+        return config_path.as_posix()
+
+    @cached_property
+    def test_conf(self):
+        """Defines the test_conf used by the evaluation pipeline.
+        """
+        test_conf = copy.deepcopy(self.configs.get("dataset_conf", {}))
+        if not "filter_conf" in test_conf:
+            test_conf["filter_conf"] = {}
+
+        if "mfcc_conf" in test_conf:
+            test_conf["mfcc_conf"]["dither"] = 0.0
+        elif not "fbank_conf" in test_conf:
+            test_conf["fbank_conf"] = {
+                "num_mel_bins": 80,
+                "frame_shift": 10,
+                "frame_length": 25,
+            }
+        if "fbank_conf" in test_conf:
+            test_conf["fbank_conf"]["dither"] = 0.0
+
+        return test_conf
+
+    @lru_cache
+    def compute_feats(
+        self,
+        audio_file: str,
+        resample_rate: int = 16000,
+        num_mel_bins=23,
+        frame_length=25,
+        frame_shift=10,
+        dither=0.0,
+    ) -> torch.Tensor:
+        waveform, sample_rate = torchaudio.load(audio_file, normalize=False)
+        logging.info(f"detected sample rate: {sample_rate}")
+        waveform = waveform.to(torch.float)
+        if sample_rate != resample_rate:
+            waveform = torchaudio.transforms.Resample(
+                orig_freq=sample_rate, new_freq=resample_rate
+            )(waveform)
+        waveform = waveform.to(self.device)
+        feats = kaldi.fbank(
+            waveform,
+            num_mel_bins=num_mel_bins,
+            frame_length=frame_length,
+            frame_shift=frame_shift,
+            dither=dither,
+            energy_floor=0.0,
+            sample_frequency=resample_rate,
+        )
+        feats = feats.unsqueeze(0)
+        return feats
+
+    def feats_batcher(
+        self,
+        infeats: torch.Tensor,
+        chunk_size: int,
+        batch_size: int,
+    ):
```

Comment:
```suggestion
    ) -> Generator[Tuple[torch.Tensor, torch.Tensor], None, None]:
```
Would need to import `Generator` from `typing`